core/src/main/scala/com/softwaremill/react/kafka_api_opt1/Consumer.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package com.softwaremill.react.kafka_api_opt1
+
+import scala.collection.immutable
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import akka.NotUsed
+import akka.stream.Graph
+import akka.stream.Inlet
+import akka.stream.Outlet
+import akka.stream.Shape
+import akka.stream.SourceShape
+import akka.stream.scaladsl.Flow
+import akka.stream.scaladsl.GraphDSL
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.clients.consumer.OffsetAndMetadata
+import org.apache.kafka.common.TopicPartition
+
+object Consumer {
+
+  // FIXME those should probably be real case classes (or value classes)?
+  type CommitMsg = Map[TopicPartition, OffsetAndMetadata]
+  type CommitConfirmation = Future[CommitMsg]
+
+  case class ConsumerShape[K, V](
+      commit: Inlet[CommitMsg],
+      messages: Outlet[ConsumerRecord[K, V]],
+      confirmation: Outlet[CommitConfirmation]
```

Comment:
Well, maybe that helps. But my general confusion is that I see no value in the commit confirmation at all. There are two scenarios:
 - the commit confirmation is not important. In that case why do we need it.
 - the commit confirmation is important. In this case either its contents are important or the fact of its reception. If the contents are important, then they must be persisted otherwise they will be lost during a crash. If the contents are not important but the reception is, then I don't see what you can do if you don't get a confirmation, i.e. then it adds no new layer of safety.

Altogether, I don't see how this commit confirmation helps here, but maybe I miss something. Maybe others can help here @patriknw @rkuhn @ktoso?